src/csharp/Grpc.Core/Internal/DebugErrorException.cs

```diff
@@ -0,0 +1,35 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Grpc.Core.Utils;
+
+namespace Grpc.Core.Internal
+{
+    /// <summary>
+    /// Represents error details provides by C-core's debug_error_string
+    /// </summary>
+    internal class DebugErrorException : Exception
```

Comment:
Does this provide any value over just using `Exception`?

Reply:
I think throwing a raw Exception is generally discouraged.
The exception provides at least some classification of the problem so it doesn't do any harm to throw a type specific exception.